gigantum/gigantum-client
# #1474: Update auth workflow to use token service

## Pull Request for Issue #1474

**Describe changes introduced by this PR (Bugs fixed, features added, docs updated, etc.)**:
This PR makes changes to the auth workflow in the client to use the new token service. This improves security while exchanging tokens from a server and generalizes the auth system so we can support many different authentication providers and servers. Client configuration has been updated to handle new fields in the well known service and a new token exchange process has been added.

**Please check if the PR fulfills these requirements**

  - PR is set to merge to integration
  - PR title is descriptive, as this becomes the commit message on squash and merge.
  - Any related issues are added to the extended commit message during squash and merge.
  - Tests have been added (for bug fixes/features)
  - Docs have been added/updated (for bug fixes/features)

**QA Test Steps and Exit Criteria**:
- Should be able to log into the client using gigantum hub
- Should be able to log out of the client and switch users using gigantum hub
- Should be able to log into the client using a team server
- Should be able to log out of the client and switch users using a team server
- Should be able to publish and sync projects and datasets using gigantum hub
- Should be able to publish and sync projects and datasets using a team server
